### PR TITLE
Make Firebase Performance initialize using Component system.

### PR DESCRIPTION
**Pull request by visumickey**
#no-changelog

Initialize Firebase Performance using Component system. This is a precursor to Sessions Library Integration.

### CODE REVIEW

File: FirebasePerformance/Sources/FPRClient.m [hunk 1]
```diff
@@ -79,17 +87,18 @@ + (void)load {
...
-    listener = nil;
+    *isCacheable = YES;
+
+    return [self sharedInstance];
   };
 
-  // Register the Perf library for Firebase Core tracking.
-  [FIRApp registerLibrary:@"fire-perf"  // From go/firebase-sdk-platform-info
-              withVersion:[NSString stringWithUTF8String:kFPRSDKVersion]];
-  listener = [notificationCenter addObserverForName:kFIRAppReadyToConfigureSDKNotification
```
Comment (leotianlizhan) [inline FirebasePerformance/Sources/FPRClient.m:89]: Not familiar with this but purely based on diff, with the `kFIRAppReadyToConfigureSDKNotification` removed, I'm guessing this is the same behavior replaced by `instantiationTiming:FIRInstantiationTimingEagerInDefaultApp` when registering components, is that right?

**Review by samedson — APPROVED** [review]
LGTM

File: FirebasePerformance/Sources/FPRClient.m [hunk 2]
```diff
@@ -48,17 +52,21 @@ @interface FPRClient ()
...
-  __block id listener;
-
-  void (^observerBlock)(NSNotification *) = ^(NSNotification *aNotification) {
-    NSDictionary *appInfoDict = aNotification.userInfo;
-    NSNumber *isDefaultApp = appInfoDict[kFIRAppIsDefaultAppKey];
-    if (![isDefaultApp boolValue]) {
-      return;
+  [FIRApp registerInternalLibrary:[FPRClient class]
+                         withName:@"fire-perf"
+                      withVersion:[NSString stringWithUTF8String:kFPRSDKVersion]];
```
Comment (paulb777) [inline FirebasePerformance/Sources/FPRClient.m:57]: Can kFPRSDKVersion ever be different than Firebase Version? I think this line should be deleted an kFPRSDKVersion be removed.

File: FirebasePerformance/Sources/FPRClient.m [hunk 3]
```diff
@@ -48,17 +52,21 @@ @interface FPRClient ()
...
+      ^id _Nullable(FIRComponentContainer *container, BOOL *isCacheable) {
+    if (!container.app.isDefaultApp) {
+      return nil;
     }
 
-    NSString *appName = appInfoDict[kFIRAppNameKey];
+    NSString *appName = container.app.name;
     FIRApp *app = [FIRApp appNamed:appName];
     FIROptions *options = app.options;
     NSError *error = nil;
```
Comment (paulb777) [inline FirebasePerformance/Sources/FPRClient.m:72]: Typo at line 74. dispatched

**Review by paulb777 — COMMENTED** [review]
Add tests like https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseRemoteConfig/Tests/Unit/FIRRemoteConfigComponentTest.m

File: FirebasePerformance/Sources/FPRClient.m (hunk 3 above)
Comment (visumickey) [reply FirebasePerformance/Sources/FPRClient.m:72]: Done.

File: FirebasePerformance/Sources/FPRClient.m (hunk 2 above)
Comment (visumickey) [reply FirebasePerformance/Sources/FPRClient.m:57]: Reference to the constant value of SDK version https://github.com/firebase/firebase-ios-sdk/blob/master/FirebasePerformance/Sources/Common/FPRConstants.m#L22

**Review by leotianlizhan — APPROVED** [review]

**Comment by visumickey** [comment]
> Add tests like https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseRemoteConfig/Tests/Unit/FIRRemoteConfigComponentTest.m

Done. Added relevant tests for the component registration and successful initialization.

**Review by paulb777 — APPROVED** [review]

**Review by ncooke3 — APPROVED** [review]

**Pull request merged**